data-pipeline/src/covid_variant_pipeline/assign_clades.py

```diff
@@ -123,14 +126,27 @@ def assign_clades(reference_tree: str):
     return assignment_file
 
 
-def main():
-    logger.info("Starting pipeline")
+@click.command()
+@click.option(
+    "--as-of-date",
+    type=click.DateTime(formats=["%Y-%m-%d"]),
+    prompt="Reference tree as of (YYYY-MM-DD)",
+    required=True,
+    help="Reference tree date to use for clade assignments (YYYY-MM-DD format)",
+)
+def main(as_of_date: str):
+    # TODO: do we need additional date validations?
+
+    # incoming as_of_date comes in as a datetime object (for validation purposes), convert back to string now
+    as_of_date = as_of_date.strftime("%Y-%m-%d")
+    logger.info("Starting pipeline", as_of_date=as_of_date)
+
     get_sequences()
     get_sequence_metadata()
-    reference_tree_file = get_reference_tree()
-    assignment_file = assign_clades(reference_tree_file)
+    reference_tree_path, root_sequence_path = save_reference_info(as_of_date)
```

Comment:
Until we're further along, I've been saving the intermediate outputs to disk (though not committing them, though I could add them to the repo if it would be helpful)

Reply:
I support not committing stuff for now.  We need to sort out where everything will live, and at the moment I don't think we're creating any outputs we'll want/need in the long term.